src/main/java/org/opentripplanner/middleware/models/OtpUser.java

```diff
@@ -68,7 +69,8 @@ public boolean delete(boolean deleteAuth0User) {
             }
         }
 
-        if (deleteAuth0User) {
+        // Only attempt to delete Auth0 user if Otp user is not assigned to third party.
+        if (deleteAuth0User && applicationId.isEmpty()) {
```

Comment:
Might need to improve this isThirdParty check (possible use for aforementioned method?).

Reply:
Again, I'm not sure this check will hold true once we add ApiUsers for the MOD UI.

Reply:
Updated to check with Auth0 if the user exists: `Auth0Users.getUserByEmail(email, false)`